shared/src/test/java/org/odk/collect/shared/geometry/GeometryTest.kt

```diff
@@ -214,31 +218,24 @@ class GeometryTest {
             generator = getTraceGenerator()
         ) { trace, intersects ->
             // Check intersects is consistent when trace is reversed
-            val reversedTrace = Trace(trace.points.reversed())
+            val reversedTrace = trace.reverse()
             assertThat(
                 "Expected intersects=$intersects:\n$reversedTrace",
                 reversedTrace.intersects(),
                 equalTo(intersects)
             )
 
             // Check intersects is consistent when trace is scaled
-            val scaleFactor = Random.nextDouble(0.1, 10.0)
-            val scaledTrace = Trace(trace.points.map {
-                Point(it.x * scaleFactor, it.y * scaleFactor)
-            })
+            val scaledTrace = trace.scale(Random.nextDouble(0.1, 10.0))
             assertThat(
                 "Expected intersects=$intersects:\n$scaledTrace",
                 scaledTrace.intersects(),
                 equalTo(intersects)
             )
 
             // Check adding an intersection makes intersects true
-            if (!intersects) {
-                val intersectionSegment = trace.segments().random()
-                val intersectPosition = Random.nextDouble(0.1, 1.0)
-                val intersectionPoint = intersectionSegment.interpolate(intersectPosition)
-                val intersectingTrace =
-                    Trace(trace.points + listOf(trace.points.last(), intersectionPoint))
+            if (!intersects && !trace.isClosed()) {
```

Comment:
Why do we skip this step for closed polygons? We can still add another segment to make sure `#intersects` returns true.